music21/roman.py

```diff
@@ -1980,86 +2063,45 @@ def adjustMinorVIandVIIByQuality(self, useScale):
-        >>> vi('VI', roman.Minor67Default.SHARP)
-        'A C# E'
-
-        For FLAT assumes lowered ^6 no matter what, while SHARP assumes raised
-        ^6 no matter what.  So #vi is needed in FLAT and bVI is needed in SHARP
-
-        >>> vi('#vi', roman.Minor67Default.FLAT)
-        'A C E'
-        >>> vi('bVI', roman.Minor67Default.SHARP)
-        'A- C E-'
-
-
-        CAUTIONARY ignores the `#` in #vi and the `b` in bVI:
-
-        >>> vi('#vi', roman.Minor67Default.CAUTIONARY)
-        'A C E'
-        >>> vi('vi', roman.Minor67Default.CAUTIONARY)
-        'A C E'
-        >>> vi('bVI', roman.Minor67Default.CAUTIONARY)
-        'A- C E-'
-        >>> vi('VI', roman.Minor67Default.CAUTIONARY)
-        'A- C E-'
-
-        Whereas QUALITY is closer to what a computer would produce, since vi is already
-        sharpened, #vi raises it even more.  And since VI is already flattened, bVI lowers
-        it even further:
-
-        >>> vi('vi', roman.Minor67Default.QUALITY)
-        'A C E'
-        >>> vi('#vi', roman.Minor67Default.QUALITY)
-        'A# C# E#'
-        >>> vi('VI', roman.Minor67Default.QUALITY)
-        'A- C E-'
-        >>> vi('bVI', roman.Minor67Default.QUALITY)
-        'A-- C- E--'
-
-        To get these odd chords with CAUTIONARY, add another sharp or flat.
-
-        >>> vi('##vi', roman.Minor67Default.CAUTIONARY)
-        'A# C# E#'
-        >>> vi('bbVI', roman.Minor67Default.CAUTIONARY)
-        'A-- C- E--'
-
-
-        For other odd chords that are contrary to the standard minor interpretation
-        in the "wrong" direction, the interpretation is the same as `QUALITY`
-
-        a major triad on raised 6?
-
-        >>> vi('#VI', roman.Minor67Default.QUALITY)
-        'A C# E'
-        >>> vi('#VI', roman.Minor67Default.CAUTIONARY)
-        'A C# E'
-
-        a minor triad on lowered 6?
-
-        >>> vi('bvi', roman.Minor67Default.QUALITY)
-        'A- C- E-'
-        >>> vi('bvi', roman.Minor67Default.CAUTIONARY)
-        'A- C- E-'
+    def _adjustMinorVIandVIIByQuality(self, workingFigure, useScale) -> str:
         '''
-        def sharpen():
+        Fix minor vi and vii to always be #vi and #vii if `.caseMatters`.
+
+        Made private in v.6.4 when `workingFigure` was added to the signature
+        and returned.
+
+        Altering `workingFigure` became necessary to handle these chromatic figures:
+        https://github.com/cuthbertLab/music21/issues/437
+
+        >>> rn = roman.RomanNumeral('viio#6', 'a')
+        >>> ' '.join([p.name for p in rn.pitches])
+        'B D G#'
+        >>> rn = roman.RomanNumeral('viio6#4', 'a')
+        >>> ' '.join([p.name for p in rn.pitches])
+        'D G# B'
+        >>> rn = roman.RomanNumeral('viio4#2', 'a')
+        >>> ' '.join([p.name for p in rn.pitches])
+        'F G# B D'
+        >>> rn = roman.RomanNumeral('viio#853', 'a')
+        >>> ' '.join([p.name for p in rn.pitches])
+        'G# B D'
+        '''
+        def sharpen(workingFigure):
```

Comment:
this is going to give some errors in my PyCharm linter which doesn't like internal variables with the same name as enclosing variables.  But I can fix it later.